packages/react-reconciler/src/ReactFiberComponentStack.js

```diff
@@ -78,3 +82,104 @@ export function getStackByFiberInDevAndProd(workInProgress: Fiber): string {
     return '\nError generating stack: ' + x.message + '\n' + x.stack;
   }
 }
+
+function describeFunctionComponentFrameWithoutLineNumber(fn: Function): string {
+  // We use this because we don't actually want to describe the line of the component
+  // but just the component name.
+  const name = fn ? fn.displayName || fn.name : '';
+  return name ? describeBuiltInComponentFrame(name) : '';
+}
+
+export function getOwnerStackByFiberInDev(workInProgress: Fiber): string {
+  if (!enableOwnerStacks || !__DEV__) {
+    return '';
+  }
```

Comment:
Are we going to fallback to `getStackByFiberInDevAndProd` in PROD?

Reply:
No, we could really error here because this should not be called in prod. Hence `InDev` suffix.

That's because we never add component stacks to prod automatically:

- We only include our own `console.error` override in DEV.
- We only include our component stacks for default onCaughtError/onUncaughtError in DEV.
- When we eventually expose this to the errorInfo object it would be on a DEV-only property.

The React DevTools could in principle be different. Basically older React versions will look the same as prod versions of new React.